connlib: Refine callbacks

francesca64 opened the pull request:
This follows-up on the discussion in #1744 and brings connlib in line with the callback revisions outlined in firezone/product#586

(It also adds some logging to the Apple bridge that was helpful when testing this)

francesca64 committed:
connlib: Callback refinements

conectado reviewed (approved):
LGTM, just need to fix the CI by skipping over the callbacks arguments in the `instrument` attribute for the new signatures.

francesca64 committed:
connlib: Update Apple callback handler + more logging

Co-authored-by: Roopesh Chander <roop@roopc.net>

francesca64 committed:
connlib: Fix Linux + Win compilation

francesca64 committed:
Merge branch 'main' into feat/connlib-refine-callbacks

francesca64 committed:
connlib: Stub `on_remove_route`

`ResourceRemoved` is currently unhandled, so this is never actually
actually called.

francesca64 committed:
connlib: Pass DNS address to `on_set_interface_config`

jamilbk committed:
Merge branch 'main' into feat/connlib-refine-callbacks
